Grbl_Esp32/src/MotionControl.cpp

```diff
@@ -313,59 +313,24 @@ void mc_homing_cycle(uint8_t cycle_mask) {
     else
 #endif
     {
-        // Search to engage all axes limit switches at faster homing seek rate.
-        if (!axis_is_squared(HOMING_CYCLE_0)) {
-            limits_go_home(HOMING_CYCLE_0);  // Homing cycle 0
-        } else {
-            ganged_mode           = SquaringMode::Dual;
-            n_homing_locate_cycle = 0;  // don't do a second touch cycle
-            limits_go_home(HOMING_CYCLE_0);
-            ganged_mode           = SquaringMode::A;
-            n_homing_locate_cycle = NHomingLocateCycle;  // restore to default value
-            limits_go_home(HOMING_CYCLE_0);
-            ganged_mode = SquaringMode::B;
-            limits_go_home(HOMING_CYCLE_0);
-            ganged_mode = SquaringMode::Dual;  // always return to dual
-        }
-#ifdef HOMING_CYCLE_1
-        if (!axis_is_squared(HOMING_CYCLE_1)) {
-            limits_go_home(HOMING_CYCLE_1);
-        } else {
-            ganged_mode           = SquaringMode::Dual;
-            n_homing_locate_cycle = 0;  // don't do a second touch cycle
-            limits_go_home(HOMING_CYCLE_1);
-            ganged_mode           = SquaringMode::A;
-            n_homing_locate_cycle = NHomingLocateCycle;  // restore to default value
-            limits_go_home(HOMING_CYCLE_1);
-            ganged_mode = SquaringMode::B;
-            limits_go_home(HOMING_CYCLE_1);
-            ganged_mode = SquaringMode::Dual;  // always return to dual
-        }
-#endif
-#ifdef HOMING_CYCLE_2
-        if (!axis_is_squared(HOMING_CYCLE_2)) {
-            limits_go_home(HOMING_CYCLE_2);
-        } else {
-            ganged_mode           = SquaringMode::Dual;
-            n_homing_locate_cycle = 0;  // don't do a second touch cycle
-            limits_go_home(HOMING_CYCLE_2);
-            ganged_mode           = SquaringMode::A;
-            n_homing_locate_cycle = NHomingLocateCycle;  // restore to default value
-            limits_go_home(HOMING_CYCLE_2);
-            ganged_mode = SquaringMode::B;
-            limits_go_home(HOMING_CYCLE_2);
-            ganged_mode = SquaringMode::Dual;  // always return to dual
+        for (uint8_t cycle = 0; cycle < MAX_N_AXIS; cycle++) {
```

Comment:
Since cycle is just a counter and isn't stored in memory, there is no value in declaring it to have a small size.  int is the basic type for an array index in C++.

Reply:
Will do